### PR TITLE
Dont hyperlink annotation timestamps when the annotation has no href

### PR DESCRIPTION
**Pull request by seanh**
This is part of the fix for <https://github.com/hypothesis/client/issues/612>. The other part is <https://github.com/hypothesis/h/pull/4711>.

A first-party annotation from the annotation read or search API looks like:

```json
{
  ...,
  links: {
    json: "http://localhost:5000/api/annotations/QvplEtX1EeeOhK9ondpFvw",
    html: "http://localhost:5000/a/QvplEtX1EeeOhK9ondpFvw",
    incontext: "http://localhost:8000/QvplEtX1EeeOhK9ondpFvw/localhost:3000/"
  },
  ...,
}
```

Notice the `html` link. A third-party annotation has no `html` or `incontext` link:

```json
{
  ...,
  links: {
    json: "http://localhost:5000/api/annotations/QvplEtX1EeeOhK9ondpFvw",
  },
  ...,
}
```

Or at least, once I've fixed h to not include these links, they won't.

With this PR the client renders the timestamps of first-party annotations as:

```html
<div ng-switch="vm.href">
  <a ng-switch-default
     target="_blank"
     title="27 Feb"
     href="https://hypothes.is/...">
       2 months ago
  </a>
</div>
```

And third-party ones as:

```html
<div ng-switch="vm.href">
  <span ng-switch-when="">
    2 months ago
  </span>
</div>
```

Annotations that are newly-created in the client and not saved to the server yet get the `<span>` version too.

### CODE REVIEW

File: src/sidebar/templates/timestamp.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<div ng-switch="vm.href">
+  <span ng-switch-when="">
```
Comment (robertknight) [inline src/sidebar/templates/timestamp.html]: The `class` attribute is missing here so the font size and color does not match between linked vs. unlinked timestamps.

Also the title attribute is missing so the full annotation creation time won't be shown on hover.

Comment (seanh) [reply src/sidebar/templates/timestamp.html]: Nice catch, cheers

**Comment by seanh** [comment]
@robertknight I've fixed the class and title. There's a 3 things duplicated now between the `<a>` and the `<span>` but I don't off the top of my head know a nice way to avoid that. I think I'd be happy to merge as-is if you are

**Comment by robertknight** [comment]
> There's 3 things duplicated now between the <a> and the <span> but I don't off the top of my head know a nice way to avoid that.

Switching element type's dynamically in Angular 1.x templates or sharing a set of attributes between elements is not trivial as far as I know.

**Pull request merged**